### PR TITLE
Inherit directly from DataClassJsonMixin instead of using @dataclass_json for improved static type checking

### PR DESCRIPTION
**Pull request by ringohoffman**
When we use the `dataclass_json` decorator, the dataclass constructors are obscured, and we get pyright errors just instantiating them:

```python
import flytekit.configuration as flyte_config

# Error: Argument missing for parameter "cls"
# Error: Expected no arguments to "DataClassJsonMixin" constructor
fast_serialization_settings = flyte_config.FastSerializationSettings(
    enabled=True,
    destination_dir=".",
    distribution_location=".",
)

```

[As it says in the dataclasses-json README](https://github.com/lidatong/dataclasses-json/blob/89578cb9ebed290e70dba8946bfdb68ff6746755/README.md?plain=1#L111-L129), we can inherit from `dataclasses_json.DataClassJsonMixin` instead for improved static type checking.

# TL;DR
This one change eliminates 467 of the 1715 existing pyright errors (27.2%) from the `flytekit` module.

## Type
 - [X] Bug Fix
 - [ ] Feature
 - [ ] Plugin

## Are all requirements met?

 - [X] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
```python
@dataclass_json
@dataclass
class EntrypointSettings:
    ...
```

now becomes

```python
@dataclass
class EntrypointSettings(DataClassJsonMixin):
    ...
```

And now there are no longer any pyright errors when instantiating any dataclasses.

### CODE REVIEW

File: flytekit/configuration/__init__.py [hunk 1]
```diff
@@ -682,9 +679,8 @@ class EntrypointSettings(object):
     path: Optional[str] = None
 
 
-@dataclass_json
 @dataclass
-class FastSerializationSettings(object):
+class FastSerializationSettings(DataClassJsonMixin):
```
Comment (kumare3) [inline flytekit/configuration/__init__.py:683]: This is cool - thank you

**Review by kumare3 — COMMENTED** [review]
Lgtm. Cc @eapolinario

**Review by wild-endeavor — APPROVED** [review]
